numpy/core/arrayprint.py

```diff
@@ -669,28 +669,29 @@ def _formatArray(a, format_function, rank, max_line_len, next_line_prefix,
     else:
         s = '['
         sep = separator.rstrip()
-        line_sep = '\n'*max(rank-1, 1)
```

Comment:
No need, we already know that `rank` is not `0` or `1`, and if it's any other value `< 1` we have bigger problems.